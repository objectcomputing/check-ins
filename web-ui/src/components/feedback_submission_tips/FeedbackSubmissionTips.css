.sbi-tips-page {
    height: 100%;
}

.header {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    grid-gap: 10px;
    align-items: center;
    margin-top: 3em;
}

.submission-tips {
    display: flex;
    flex-direction: row;
    margin: 8em 4em 0 4em;
    justify-content: space-evenly;
}

.sbi-circle {
    background: #3f51b5;
    color: #fff;
    width: 250px;
    height: 250px;
    line-height: 250px;
    border-radius: 50%;
    font-size: 150px;
    text-align: center;
}

.sbi-info {
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
    padding: 0 2em 0 2em;
    max-width: 400px;
}

.sbi-info .sbi-info-title {
    margin: 20px 0 10px 0;
}

@media (max-width: 800px) {

    .header {
        display: grid;
        grid-template-columns: 1fr 1fr;
        grid-gap: 10px;
        align-items: center;
        margin-top: 3em;
        padding-left: 14px;
        margin-bottom: 28px;
    }

    .submission-tips {
        display: flex;
        flex-direction: column;
        margin: 2em 1em 1em 1em;
    }

    .sbi-info {
        display: flex;
        flex-direction: row;
        padding: 0;
        margin-bottom: 2em;
    }

    .sbi-circle {
        width: 80px;
        height: 80px;
        font-size: 30px;
        line-height: 80px;
        min-width: 80px;
        margin-right: 20px;
    }

    .sbi-info .sbi-info-title {
        font-size: 20px;
        text-align: left;
    }

    .sbi-info .sbi-info-description {
        font-size: 12px;
        text-align: left;
    }
}