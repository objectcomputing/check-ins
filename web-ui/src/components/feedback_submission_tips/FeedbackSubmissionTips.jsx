import React from "react";
import Typography from "@material-ui/core/Typography";
import {makeStyles} from '@material-ui/core/styles';

import "./FeedbackSubmissionTips.css";
import Button from "@material-ui/core/Button";
import PropTypes from "prop-types";

const useStyles = makeStyles({
  title: {
    textAlign: "center",
    gridColumn: 2,
    ['@media (max-width: 800px)']: { // eslint-disable-line no-useless-computed-key
      fontSize: "34px",
      gridColumn: 1,
      textAlign: "left",

    }
  },

  button: {
    justifySelf: "end",
    marginRight: "3em",
    ['@media (max-width: 800px)']: { // eslint-disable-line no-useless-computed-key
      marginRight: "0",
    }
    },

  announcement: {
    textAlign: "center",
    marginBottom: "0px",
    ['@media (max-width: 800px)']: { // eslint-disable-line no-useless-computed-key
      fontSize: "20px",
    }
  }
});

const sbiInfo = [
  {
    letter: "S",
    title: "Situation",
    description: "Describe the situation; be specific about when and where it occurred."
  },
  {
    letter: "B",
    title: "Behavior",
    description: "Describe the observable behavior; don't assume you know what the other person was thinking."
  },
  {
    letter: "I",
    title: "Impact",
    description: "Describe what you thought and felt as a result of that behavior."
  }
];

const propTypes = {
  onNextClick: PropTypes.func
}

const FeedbackSubmissionTips = (props) => {
  const classes = useStyles();
  return (
    <div className="sbi-tips-page">
      <div className="header">
        <Typography className={classes.title} variant="h2"><b>Feedback Tips</b></Typography>
        <div className={classes.button}>
          <Button
            onClick={() => props.onNextClick()}
            variant="contained"
            color="primary">
            Next
          </Button>
        </div>
      </div>
      <Typography className={classes.announcement} variant="h5"><b>SBI is a common approach to providing constructive and fair feedback </b></Typography>
      <div className="submission-tips">
        {sbiInfo.map((info) => (
          <div className="sbi-info" key={info.letter}>
            <div className="sbi-circle">{info.letter}</div>
            <div className="sbi-title-and-description">
              <Typography className="sbi-info-title" variant="h3">{info.title}</Typography>
              <Typography className="sbi-info-description" variant="body1">{info.description}</Typography>
            </div>
          </div>
        ))}
      </div>
    </div>
  );
}

FeedbackSubmissionTips.propTypes = propTypes;

export default FeedbackSubmissionTips;