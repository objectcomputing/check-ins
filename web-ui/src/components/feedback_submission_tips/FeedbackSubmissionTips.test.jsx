import React from "react";
import FeedbackSubmissionTips from "./FeedbackSubmissionTips";

it("renders the feedback tips component.", () => {
  snapshot(<FeedbackSubmissionTips/>);
});