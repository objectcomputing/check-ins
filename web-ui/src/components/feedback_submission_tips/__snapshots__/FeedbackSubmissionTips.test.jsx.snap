// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders the feedback tips component. 1`] = `
<div>
  <div
    class="sbi-tips-page css-y8zz4t"
  >
    <div
      class="submission-tips-header"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 FeedbackSubmissionTips-title css-1sra7t5-MuiTypography-root"
      >
        <b>
          Feedback Tips
        </b>
      </h2>
      <div
        class="FeedbackSubmissionTips-button"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Next
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <h5
      class="MuiTypography-root MuiTypography-h5 FeedbackSubmissionTips-announcement css-ag7rrr-MuiTypography-root"
    >
      <b>
        SBI is a common approach to providing constructive and fair feedback 
      </b>
    </h5>
    <div
      class="submission-tips"
    >
      <div
        class="sbi-info"
      >
        <div
          class="sbi-circle"
        >
          S
        </div>
        <div
          class="sbi-title-and-description"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 sbi-info-title css-gepadz-MuiTypography-root"
          >
            Situation
          </h3>
          <p
            class="MuiTypography-root MuiTypography-body1 sbi-info-description css-ahj2mt-MuiTypography-root"
          >
            Describe the situation; be specific about when and where it occurred.
          </p>
        </div>
      </div>
      <div
        class="sbi-info"
      >
        <div
          class="sbi-circle"
        >
          B
        </div>
        <div
          class="sbi-title-and-description"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 sbi-info-title css-gepadz-MuiTypography-root"
          >
            Behavior
          </h3>
          <p
            class="MuiTypography-root MuiTypography-body1 sbi-info-description css-ahj2mt-MuiTypography-root"
          >
            Describe the observable behavior; don't assume you know what the other person was thinking.
          </p>
        </div>
      </div>
      <div
        class="sbi-info"
      >
        <div
          class="sbi-circle"
        >
          I
        </div>
        <div
          class="sbi-title-and-description"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 sbi-info-title css-gepadz-MuiTypography-root"
          >
            Impact
          </h3>
          <p
            class="MuiTypography-root MuiTypography-body1 sbi-info-description css-ahj2mt-MuiTypography-root"
          >
            Describe what you thought and felt as a result of that behavior.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
